test-it/karma/publicapi/AllocateTypes.Test.js

```diff
@@ -0,0 +1,254 @@
+describe('Vireo public API allows', function () {
+    'use strict';
+    // Reference aliases
+    var Vireo = window.NationalInstruments.Vireo.Vireo;
+    var vireoRunner = window.testHelpers.vireoRunner;
+    var fixtures = window.testHelpers.fixtures;
+
+    var vireo = new Vireo();
+    var internalEggShell = vireo.eggShell.internal_module_do_not_use_or_you_will_be_fired.eggShell;
+
+    var publicApiAllocateTypesViaUrl = fixtures.convertToAbsoluteFromFixturesDir('publicapi/AllocateTypes.via');
+    var viName = 'AllocateTypes';
+
+    var expectValidValueRef = function (valueRef) {
+        expect(valueRef).toBeNonEmptyObject();
+        expect(valueRef.typeRef).toBeNumber();
+        expect(valueRef.typeRef).not.toBe(0);
+        expect(valueRef.dataRef).toBeNumber();
+        expect(valueRef.dataRef).not.toBe(0);
+    };
+
+    var allocateData = function (viName, path) {
+        var valueRef = vireo.eggShell.findValueRef(viName, path);
+        return internalEggShell.allocateData(valueRef.typeRef);
+    };
+
+    var writeValue = function (valueRef, value) {
+        vireo.eggShell.writeJSON(valueRef, JSON.stringify(value));
+    };
+
+    var readValue = function (valueRef) {
+        return JSON.parse(vireo.eggShell.readJSON(valueRef));
+    };
+
+    var allocateTest = function (viName, path, value) {
+        var dataValueRef = allocateData(viName, path);
+        expectValidValueRef(dataValueRef);
+        writeValue(dataValueRef, value);
+        expect(readValue(dataValueRef)).toEqual(value);
+        internalEggShell.deallocateData(dataValueRef);
+    };
+
+    beforeAll(function (done) {
+        fixtures.preloadAbsoluteUrls([
+            publicApiAllocateTypesViaUrl
+        ], done);
+    });
+
+    beforeAll(function () {
+        vireoRunner.rebootAndLoadVia(vireo, publicApiAllocateTypesViaUrl);
+    });
+
+    var tryAllocate = function (typeRef) {
+        return function () {
+            internalEggShell.allocateData(typeRef);
+        };
+    };
+
+    var tryDeallocate = function (valueRef) {
+        return function () {
+            internalEggShell.deallocateData(valueRef);
+        };
+    };
+
+    describe('error handling', function () {
+        describe('for allocateData throws', function () {
+            it('when typeRef is invalid', function () {
+                var invalidTypeRef = 1234;
+
+                expect(tryAllocate(invalidTypeRef)).toThrowError(/InvalidTypeRef/);
+            });
+        });
+
+        describe('for deallocateData throws', function () {
+            it('when typeRef is invalid', function () {
```

Comment:
From what I understand of this kind of testing framework, the reason the function is called "it" is so the test reads like an english sentence. In that case, please use strings here that read like a sentence, e.g. it('throws an error when given an invalid value ref.')

Reply:
I agree with you. I was more worried about how the whole test name ends up in the output and not so much with the readability of each `it()`. I'll update this. Thanks for pointing it out.